openbb_terminal/common/behavioural_analysis/twitter_view.py

```diff
@@ -152,52 +153,111 @@ def display_sentiment(
     df_tweets = df_tweets.sort_values(by="date")
     df_tweets["cumulative_compound"] = df_tweets["sentiment"].cumsum()
 
-    # This plot has 2 axis
-    if external_axes is None:
-        _, axes = plt.subplots(
-            2, 1, sharex=True, figsize=plot_autoscale(), dpi=cfg_plot.PLOT_DPI
+    # get stock end price for each corresponding day if compare == True
+    if not compare:
```

Comment:
Because the ax1 and ax2 do not change regardless of whether we display stock price or not. We should do something like the following pseudo-code:

```
if compare:
    ax1,ax2,ax3 are created
else:
   ax1,ax2 are created

populate ax1 and ax2 (this avoids repeated code)

if compare:
   populate ax3 with stock price

set theme
```

This allows to minimize duplicated code and makes it easier for debug

Reply:
I see yeah this makes sense to me, we have to just write `if compare` twice, i thought that wouldn't be good idea, but yeah we can do this.

Reply:
Makes it easier from the readability perspective and maintenance as well since there isn't repeated code